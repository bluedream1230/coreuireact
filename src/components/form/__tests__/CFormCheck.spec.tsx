import * as React from 'react'
import { render /* ,screen */ /* ,fireEvent */ /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CFormCheck } from '../../../index'

test('loads and displays CFormCheck component', async () => {
  const { container } = render(<CFormCheck />)
  expect(container).toMatchSnapshot()
})

test('CFormCheck customize button=false', async () => {
  const { container } = render(
    <CFormCheck
      button={false}
      className="bazinga"
      id="id"
      inline={true}
      label="label"
      size="xl"
      type="radio"
      switch={true}
    />,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('form-check')
  expect(container.firstChild).toHaveClass('form-switch')
  expect(container.firstChild).toHaveClass('form-switch-xl')
  expect(container.firstChild).toHaveClass('form-check-inline')
})

test('CFormCheck customize button=true', async () => {
  const { container } = render(
    <CFormCheck
      button={true}
      className="bazinga"
      buttonColor="primary"
      buttonShape="rounded-circle"
      buttonSize="lg"
      buttonVariant="ghost"
      id="id"
      inline={true}
      label="label"
      size="xl"
      type="radio"
      switch={true}
    />,
  )
  expect(container).toMatchSnapshot()
})
