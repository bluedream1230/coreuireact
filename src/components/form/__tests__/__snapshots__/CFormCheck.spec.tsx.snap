// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormCheck customize button=false 1`] = `
<div>
  <div
    class="form-check form-switch form-switch-xl form-check-inline bazinga"
  >
    <input
      class="form-check-input"
      id="id"
      type="radio"
    />
    <label
      class="form-check-label"
      for="id"
    >
      label
    </label>
  </div>
</div>
`;

exports[`CFormCheck customize button=true 1`] = `
<div>
  <div
    class="form-check form-switch form-switch-xl form-check-inline bazinga"
  >
    <input
      class="btn-check"
      id="id"
      type="radio"
    />
    <label
      class="btn btn-ghost-primary btn-lg buttonShape"
      for="id"
    >
      label
    </label>
  </div>
</div>
`;

exports[`loads and displays CFormCheck component 1`] = `
<div>
  <input
    class="form-check-input"
    type="checkbox"
  />
</div>
`;
